### ISSUE TITLE
Safer Cluster Access with IAP Bastion Host exposes cluster external endpoint

### ISSUE THREAD

**Issue by nasseredine** [terraform-google-modules/terraform-google-kubernetes-engine#1875]
### TL;DR

In the _[Safer Cluster Access with IAP Bastion Host](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/tree/v30.0.0/examples/safer_cluster_iap_bastion)_ example, the cluster external endpoint should not be enabled.

### Expected behavior

In the _[Safer Cluster Access with IAP Bastion Host](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/tree/v30.0.0/examples/safer_cluster_iap_bastion)_ example, the cluster external endpoint shouldn't be enabled.

Enabling the cluster's external endpoint negates the benefit of the Bastion host (without external IP) with an IAP access pattern to the private cluster.

### Observed behavior

The cluster's external endpoint is exposed, though not whitelisted in the `master_authorized_networks` list of authorized networks that can access the control plane.

### Terraform Configuration

```hcl
# Terraform configuration taken from the Safer Cluster Access with IAP Bastion Host example

locals {
  bastion_name = format("%s-bastion", var.cluster_name)
  bastion_zone = format("%s-a", var.region)
}

module "bastion" {
  source  = "terraform-google-modules/bastion-host/google"
  version = "~> 6.0"

  network        = module.vpc.network_self_link
  subnet         = module.vpc.subnets_self_links[0]
  project        = module.enabled_google_apis.project_id
  host_project   = module.enabled_google_apis.project_id
  name           = local.bastion_name
  zone           = local.bastion_zone
  image_project  = "debian-cloud"
  machine_type   = "g1-small"
  startup_script = templatefile("${path.module}/templates/startup-script.tftpl", {})
  members        = var.bastion_members
  shielded_vm    = "false"
}

module "gke" {
  source  = "terraform-google-modules/kubernetes-engine/google//modules/safer-cluster"
  version = "~> 29.0"

  project_id              = module.enabled_google_apis.project_id
  name                    = var.cluster_name
  region                  = var.region
  network                 = module.vpc.network_name
  subnetwork              = module.vpc.subnets_names[0]
  ip_range_pods           = module.vpc.subnets_secondary_ranges[0][0].range_name
  ip_range_services       = module.vpc.subnets_secondary_ranges[0][1].range_name
  enable_private_endpoint = false
  deletion_protection     = false
  master_authorized_networks = [{
    cidr_block   = "${module.bastion.ip_address}/32"
    display_name = "Bastion Host"
  }]
  database_encryption = [
    {
      "key_name" : module.kms.keys["gke-key"],
      "state" : "ENCRYPTED"
    }
  ]
  grant_registry_access = true
  node_pools = [
    {
      name          = "safer-pool"
      min_count     = 1
      max_count     = 4
      auto_upgrade  = true
      node_metadata = "GKE_METADATA"
    }
  ]
}
```

### Terraform Version

```sh
Terraform v1.7.1
on darwin_amd64
+ provider registry.terraform.io/hashicorp/google v5.14.0
+ provider registry.terraform.io/hashicorp/google-beta v5.14.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.25.2
+ provider registry.terraform.io/hashicorp/null v3.2.2
+ provider registry.terraform.io/hashicorp/random v3.6.0
+ provider registry.terraform.io/hashicorp/time v0.10.0

Your version of Terraform is out of date! The latest version
is 1.7.3. You can update by downloading from https://www.terraform.io/downloads.html
```

### Additional information

In the example, is there a reason behind exposing the control plane through a public IP?

The variable`enable_private_endpoint` when set to `false` enables both public and private endpoints, and only the private endpoint when set to `true` (applies only to private clusters in this case). Maybe a better name could have been `enable_private_endpoint_only`.
